containers/init_docker.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# This script builds the rfaas-base docker container, pushes it to the
+# registry, sets up a docker volume, and generates example configuration 
+# json which should then be updated in config/executor_manager.json.
+# The script also configures a docker network for suitable use with
+# docker_rdma_sriov
+
+# NOTE: Run this script from repo root, and make sure the sriov docker plugin
+# is installed
+
+set -e
+
+if [ $# -lt 3 ]; then
+    echo "usage: ./init_docker.sh <REGISTRY IP> <REGISTRY PORT> <SUBNET>"
+    exit
+fi
+
+REG_IP=$1    # IP or name of the docker registry
+REG_PORT=$2  # Port of the docker registry
+SUBNET=$3    # Subnet for the docker network
+
+IMG_NAME=rfaas-base
+REG_IMG=$REG_IP:$REG_PORT/$IMG_NAME
+
+# Build the docker container, login and push to the registry
+sudo docker build -t $IMG_NAME - < containers/rfaas-base.Dockerfile
```

Comment:
I think it might be simpler not to use `sudo` here and instead explain in the documentation that users to make sure that they have necessary permissions to run Docker. Furthermore, on clusters the user might be in the `docker` group but it it is very unlikely that they will have `sudo` access.

Also, it's safer if never request to execute anything with superuser permissions - less likely that any mistake will have bad consequences :)

Reply:
Okay. That is a great point. Principle of least privilege!